libr/include/r_bin.h

```diff
@@ -518,6 +518,8 @@ R_API void r_bin_trycatch_free(RBinTrycatch *tc);
 
 typedef struct r_bin_plugin_t {
 	RPluginMeta meta;
+	void (*init)();
+	void (*fini)();
```

Comment:
Init and fini should take rbinfile as argument because they need a context and a place to save their global state

Reply:
for bf related things there is load and destroy. init and fini are needed for plugins that need a global runtime, like in actual global scope. if you init the mono runtime twice, it will crash. so this needs a to live outside rbinfile (in the global scope of the plugin), or you could not load multiple dotnet bins

Reply:
Load and destroy sound like a very bad lifecycle names to me

Reply:
didn't you came up with these?